backend/persistent/datastores/Athena.js

```diff
@@ -0,0 +1,189 @@
+
+import {executeQuery} from './drivers/AWSAthenaDriver';
+
+import Logger from '../../logger';
+const SHOW_TABLES_QUERY = 'SHOW TABLES';
+const SHOW_SCHEMA_QUERY = 'SELECT table_name, column_name, data_type FROM '
+    + 'information_schema.columns WHERE table_schema ';
+const DEFAULT_QUERY_TIMEOUT = 2000;
+
+/*
+ * The connection function will validate the parameters and return the connection
+ * parameters
+ * @param {object} connection
+ * @param {string} connection.accessKey - AWS Access Key
+ * @param {string} connection.secretAccessKey - AWS Secret Key
+ * @param {string} connection.region - AWS Region
+ * @param {string} connection.database - Database name to connect to
+ * @param {string} connection.sqlStatement - SQL statement to execute
+ * @param {string} connection.s3Outputlocation - Location will Athena will output resutls of query
+ * @param {number} params.queryTimeout  - The timeout interval when the query should stop
+ * @returns {Promise} that resolves connection
+ */
+export function connect(connection) {
+    const {
+        region, accessKey, secretKey, database, sqlStatement, s3Outputlocation, queryTimeout
+    } = connection;
+
+    return new Promise(function(resolve, reject) {
+
+        if (!region) {
+            return reject(new Error('The AWS Region was not defined'));
+        }
+
+        if (!accessKey) {
+            return reject(new Error('The AWS access key was not defined'));
+        }
+
+        if (!secretKey) {
+            return reject(new Error('The AWS secret key was not defined'));
+        }
+
+        if (!database) {
+            return reject(new Error('The Database Name was not defined'));
+        }
+
+        if (!s3Outputlocation) {
+            return reject(new Error('The Athena S3 Results Output location was not defined'));
+        }
+
+        if (!queryTimeout && queryTimeout < 0) {
+            return reject(new Error('The Athena Query Timeout was not defined'));
+        }
+        const con = {
+            region, accessKey, secretKey, database, sqlStatement, s3Outputlocation, queryTimeout
+        };
+        resolve(con);
```

Comment:
We need to check that the connection works. How about checking this by getting the list of execution queries?